src/Components/Modals/EditModal/EditModal.tsx

```diff
@@ -39,96 +42,109 @@ export function EditModal({onClose, movie} : EditModalProps) {
-        <input className='form-input' value={form.title} placeholder="Movie title" onChange={(e) => dispatch({type: 'title', payload: e.target.value})}/>
-      </div>
-
-      <div className="second-column">
-        <label>RELEASE DATE</label>
-        <DatePicker className='form-input' selected={new Date(form.release_date)} placeholderText="Select Date" onChange={(e) => dispatch({type: 'release_date', payload: e})}/>
-      </div>
-
-      <div className="first-column">
-        <label>POSTER URL</label>
-        <input className='form-input' value={form.poster_path} placeholder="https://" onChange={(e) => dispatch({type: 'poster_path', payload: e.target.value})}/>
-      </div>
-
-      <div className="second-column">
-        <label>RATING</label>
-        <input className='form-input' value={form.vote_average || ''} placeholder="7.8" onChange={(e) => dispatch({type: 'vote_average', payload: +e.target.value})}/>
-      </div>
-
-      <div className="first-column multiselect">
-        <label>GENRE</label>
-        <MultiSelect hasSelectAll={false}
-          options={Genres.map((g) => {
-            return {label: g, value: g};
-          })}
-          value={form.genres.map((g) => {
-            return {label: g, value: g};
-          })}
-          onChange={(e) => dispatch({type: 'genres', payload: e})}
-          labelledBy="Select" />
-      </div>
-
-      <div className="second-column">
-        <label>RUNTIME</label>
-        <input className='form-input' value={form.runtime || ''} placeholder="minutes" onChange={(e) => dispatch({type: 'runtime', payload: +e.target.value})}/>
-      </div>
-
-      <div className="overview">
-        <label>OVERVIEW</label>
-        <textarea value={form.overview} placeholder="Movie description" onChange={(e) => dispatch({type: 'overview', payload: e.target.value})}/>
-      </div>
 
-      <button className='transparent-button reset-button' onClick={(e) => dispatch({type: 'reset'})}>RESET</button>
-      <button className='red-button submit-button' onClick={onSave}>SUBMIT</button>
+      <Formik
+        initialValues={initialValues}
+        validationSchema={Yup.object({
+          title: Yup.string()
+              .required('Required'),
+          release_date: Yup.date()
+              .required('Required'),
+          poster_path: Yup.string().url()
+              .required('Required'),
+          vote_average: Yup.number()
+              .required('Required')
+              .min(0)
+              .max(100),
+          genres: Yup.array()
+              .of(Yup.string())
+              .min(1)
+              .required('Required'),
+          runtime: Yup.number()
+              .required('Required')
+              .min(0),
+          overview: Yup.string()
+              .required('Required'),
+        })}
+        onSubmit={async (values) => {
+          const id = movie.id;
+          debugger;
+          await onSave({...values, id});
+        }}>
+
+        {(formProps) =>{

```

Comment:
Just note: I would think how avoid using render prop or cover it into useCallback